cannot build on dev: macOS

with https://github.com/ahrm/sioyek/issues/926#issuecomment-1902626075 in mind, still got an error:
```
In file included from pdf_viewer/coordinates.cpp:1:
pdf_viewer/coordinates.h:160:23: error: no member named 'max' in namespace 'std'; did you mean 'fmax'?
        res.x1 = std::max(R::x1, other.x1);
                 ~~~~~^~~
                      fmax
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/cmath:434:9: note: 'fmax' declared here
using ::fmax _LIBCPP_USING_IF_EXISTS;
        ^
In file included from pdf_viewer/coordinates.cpp:1:
pdf_viewer/coordinates.h:162:23: error: no member named 'max' in namespace 'std'; did you mean 'fmax'?
        res.y1 = std::max(R::y1, other.y1);
                 ~~~~~^~~
                      fmax
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/cmath:434:9: note: 'fmax' declared here
using ::fmax _LIBCPP_USING_IF_EXISTS;
        ^
```

--- a/README.md
+++ b/README.md
@@ -151,13 +151,29 @@ build_windows.bat
 ```
 
 ### Mac
-1. Install Xcode and Qt 5.
-2. Clone the repository and build:
+1. Uninstall previous Qt6 installed by Homebrew
+2. Install Xcode.
+3. Install Qt6.
+```
+pip install aqtinstall
+cd /path/to/qt
+aqt install-qt mac desktop 6.6.1 clang_64 -m all
+export Qt6_DIR=/path/to/qt/6.6.1/macos/
+export QT_PLUGIN_PATH=/path/to/qt/6.6.1/macos/plugins
+export PKG_CONFIG_PATH=/path/to/qt/6.6.1/macos/lib/pkgconfig
+export QML2_IMPORT_PATH=/path/to/qt/6.6.1/macos/qml
+export PATH="/path/to/qt/6.6.1/macos/bin:$PATH"
+```
+4. Clone the repository, build and install:
 ```
 git clone --recursive https://github.com/ahrm/sioyek
 cd sioyek
 chmod +x build_mac.sh
-./build_mac.sh
+setopt PIPE_FAIL PRINT_EXIT_VALUE ERR_RETURN SOURCE_TRACE XTRACE
+MAKE_PARALLEL=8 ./build_mac.sh
+
+mv build/sioyek.app /Applications/
+sudo codesign --force --sign - --deep /Applications/sioyek.app
 ```
 
 ## Donation

--- a/build_mac.sh
+++ b/build_mac.sh
@@ -46,4 +46,3 @@ INFO_PLIST="build/sioyek.app/Contents/Info.plist"
 
 macdeployqt build/sioyek.app -dmg
 zip -r sioyek-release-mac.zip build/sioyek.dmg
-codesign --force --deep --sign - build/sioyek.app

--- a/build_mac.sh
+++ b/build_mac.sh
@@ -34,5 +34,16 @@ cp pdf_viewer/keys.config build/sioyek.app/Contents/MacOS/keys.config
 cp pdf_viewer/keys_user.config build/sioyek.app/Contents/MacOS/keys_user.config
 cp tutorial.pdf build/sioyek.app/Contents/MacOS/tutorial.pdf
 
+# Capture the current PATH
+CURRENT_PATH=$(echo $PATH)
+
+# Define the path to the Info.plist file inside the app bundle
+INFO_PLIST="build/sioyek.app/Contents/Info.plist"
+
+# Add LSEnvironment key with PATH to Info.plist
+/usr/libexec/PlistBuddy -c "Add :LSEnvironment dict" "$INFO_PLIST" || echo "LSEnvironment already exists"
+/usr/libexec/PlistBuddy -c "Add :LSEnvironment:PATH string $CURRENT_PATH" "$INFO_PLIST" || /usr/libexec/PlistBuddy -c "Set :LSEnvironment:PATH $CURRENT_PATH" "$INFO_PLIST"
+
 macdeployqt build/sioyek.app -dmg
 zip -r sioyek-release-mac.zip build/sioyek.dmg
+codesign --force --deep --sign - build/sioyek.app

